### PR TITLE
Dev/connector lib

### PR DESCRIPTION
**Pull request by sramani-metro**

### CODE REVIEW

**Review by pwielders — DISMISSED** [review]

File: Source/connector/Connector.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+#pragma once
+
+namespace WPEFramework {
+
+namespace PluginHost {
+    struct IShell;
+}
+
+namespace Connector {
```
Comment (pwielders) [inline Source/connector/Connector.h]: Although technically there is nothing wrong with this, I would like to hide it from the external interface and force all users to the C functions, to make sure that there is only one way for accessing for these methods. Less exposure is less maintenance and less portability risk. 

So lets hide this part: https://github.com/rdkcentral/Thunder/pull/891/files#diff-276434645d979f1b866322ebbce0d324c54b0a94e89821f4f50afe7d77107b1cR28-R51 and only allow the C functions to be accessed. This class off cource can be sused interbnally to this implementation thats fine, so it is just to move the class to the cpp file...

File: Source/connector/Connector.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "Module.h"
+#include "Connector.h"
+#include "plugins/IShell.h"
+
+namespace WPEFramework {
+namespace Connector {
+
```
Comment (pwielders) [inline Source/connector/Connector.cpp:25]: Move the class declaration in here

File: Source/WPEFramework/Controller.cpp [hunk 3]
```diff
@@ -70,6 +71,9 @@ namespace Plugin {
 
         _resumes.clear();
         _service = service;
+        
+        Connector::Connector::Instance().Announce(service);
```
Comment (pwielders) [inline Source/WPEFramework/Controller.cpp]: Use the C function i.s.o. the C++ functionality, so remark for this class declaration.

File: Source/WPEFramework/Controller.cpp [hunk 4]
```diff
@@ -110,6 +114,8 @@ namespace Plugin {
     {
         ASSERT(_service == service);
 
+        Connector::Connector::Instance().Revoke(service);
```
Comment (pwielders) [inline Source/WPEFramework/Controller.cpp]: Use the C function i.s.o. the C++ functionality, so remark for this class declaration.

File: Source/plugins/Types.h [hunk 5]
```diff
@@ -267,13 +271,13 @@ namespace RPC {
         {
             ASSERT(_controller == nullptr);
 
+            _controller = Connector::Connector::Instance().Controller();
```
Comment (pwielders) [inline Source/plugins/Types.h]: Use the C function i.s.o. the C++ functionality, so remark for this class declaration.

**Review by pwielders — CHANGES_REQUESTED** [review]

File: Source/connector/Connector.cpp (hunk 2 above)
Comment (sramani-metro) [reply Source/connector/Connector.cpp:25]: Done

File: Source/WPEFramework/Controller.cpp (hunk 4 above)
Comment (sramani-metro) [reply Source/WPEFramework/Controller.cpp]: Done

File: Source/plugins/Types.h (hunk 5 above)
Comment (sramani-metro) [reply Source/plugins/Types.h]: Done

File: Source/connector/Connector.h (hunk 1 above)
Comment (sramani-metro) [reply Source/connector/Connector.h]: Done

**Review by pwielders — APPROVED** [review]

**Pull request merged**